identity/src/main/java/com/android/identity/util/ApplicationData.java

```diff
@@ -0,0 +1,165 @@
+
+    /**
+     * Sets application specific data as a string.
+     *
+     * <p>Like {@link #setData(String, byte[])} but encodes the given value
+     * using <a href="http://cbor.io/">CBOR</a> before storing it.
+     *
+     * <p> Use {@link #setData(String, byte[])} with {@code null} as the value in order to remove.
+     *
+     * @param key   the key for the data.
+     * @param value the value
+     * @return      the modified {@link ApplicationData}.
+     */
+    ApplicationData setString(@NonNull String key, @NonNull String value);
+
+    /**
+     * Sets application specific data as a {@link Number}.
+     *
+     * <p>Like {@link #setData(String, byte[])} but encodes the given value
+     * using <a href="http://cbor.io/">CBOR</a> before storing it.
+     *
+     * <p> Use {@link #setData(String, byte[])} with {@code null} as the value in order to remove.
+     *
+     * @param key   the key for the data.
+     * @param value the value.
+     * @return      the modified {@link ApplicationData}.
+     */
+    ApplicationData setNumber(@NonNull String key, long value);
+
+    /**
+     * Sets application specific data as a {@link Timestamp}.
+     *
+     * <p>Like {@link #setData(String, byte[])} but encodes the given value
+     * using <a href="http://cbor.io/">CBOR</a> before storing it.
+     *
+     * <p> Use {@link #setData(String, byte[])} with {@code null} as the value in order to remove.
+     *
+     * @param key   the key for the data.
+     * @param value the value.
+     * @return      the modified {@link ApplicationData}.
+     */
+    ApplicationData setTimestamp(@NonNull String key, @NonNull Timestamp value);
+
+    /**
+     * Sets application specific data as a boolean.
+     *
+     * <p>Like {@link #setData(String, byte[])} but encodes the given value
+     * using <a href="http://cbor.io/">CBOR</a> before storing it.
+     *
+     * <p> Use {@link #setData(String, byte[])} with {@code null} as the value in order to remove.
+     *
+     * @param key   the key for the data.
+     * @param value the value.
+     * @return      the modified {@link ApplicationData}.
+     */
+    ApplicationData setBoolean(@NonNull String key, boolean value);
+
+    /**
+     * Returns whether the {@link ApplicationData} has a value for the key provided.
+     *
+     * @param key the key for the data.
+     * @return    true if the key exists, false else.
+     */
+    boolean keyExists(String key);
+
+    /**
+     * Gets application specific data.
+     *
+     * <p>Gets data previously stored with {@link #getData(String)}.
+     *
+     * @param  key the key for the data.
+     * @return the value.
+     * @throws IllegalArgumentException if the data element does not exist.
+     */
+    @NonNull byte[] getData(@NonNull String key);
+
+    /**
+     * Gets application specific data as a string.
+     *
+     * <p>Like {@link #getData(String)} but decodes the value as
+     * an UTF-8 string before returning it.
+     *
```

Comment:
It's not clear that CBOR decoding is involved here. I think it's clearer if this says "Takes the data returned by {@link getData(String)} and decodes it as a CBOR string". You also want to add another "@throws IllegalArgumentException if the data isn't a CBOR encoded string" since that's entirely possible.

Same for all the other getters below.